src/couch_httpd_misc_handlers.erl

```diff
@@ -185,6 +185,7 @@ handle_config_req(#httpd{method='GET', path_parts=[_, Section, Key]}=Req) ->
     end;
 % POST /_config/_reload - Flushes unpersisted config values from RAM
 handle_config_req(#httpd{method='POST', path_parts=[_, <<"_reload">>]}=Req) ->
+    couch_httpd:validate_ctype(Req, "application/json"),
```

Comment:
_compact and _ensure_full_commit requests doesn't take any body as well, but they also ensures in JSON mimetype. So it's for consistent behaviour. However, need to forcely read the body `_ = couch_httpd:body(Req),` to prevent request hangs.